pkg/storage/iscsiNode.go

```diff
@@ -159,13 +237,37 @@ func (iscsi *iscsiStorage) NodeUnpublishVolume(ctx context.Context, req *csi.Nod
 	}
 
 	volumeName, _ := common.VolumeIdGetName(req.GetVolumeId())
-	klog.Infof("unpublishing volume %s at target path %s", volumeName, req.GetTargetPath())
+
+	// Ensure that NodePublishVolume and NodeUnpublishVolume are only called once per volume
+	if mu, present := gateKeepers[volumeName]; present {
```

Comment:
ditto

Reply:
Switched to using the stringlock module you provided.